python/ray/experimental/dag/dag_node.py

```diff
@@ -278,6 +278,22 @@ def __reduce__(self):
         """
         raise ValueError(f"DAGNode cannot be serialized. DAGNode: {str(self)}")
 
+    def bind(self, *args, **kwargs):
+        raise AttributeError(
+            "All DAGNode types act as intermediate representation (IR) of a "
+            "DAG that can only be produced by .bind() on supported class or "
+            "function, and executed via dag.execute(input). Please do not call "
+            f".bind() on a DAGNode IR object of type: {type(self)}"
+        )
+
+    def remote(self, *args, **kwargs):
+        raise AttributeError(
+            "All DAGNode types act as intermediate representation (IR) of a "
```

Comment:
"`.remote()` cannot be used on {node_type}. To execute the task graph for this node, use `.execute()``.